src/battle_script_commands.c

```diff
@@ -1609,22 +1609,23 @@ static bool32 AccuracyCalcHelper(u16 move)
         JumpIfMoveFailed(7, move);
         return TRUE;
     }
-    else if (GetBattlerAbility(gBattlerAttacker) == ABILITY_NO_GUARD)
+    // If the attacker has the ability No Guard and they aren't targeting a Pokemon involved in a Sky Drop with the move Sky Drop, move hits.
+    else if (GetBattlerAbility(gBattlerAttacker) == ABILITY_NO_GUARD && (move != MOVE_SKY_DROP || (gBattlerTarget != gBattleStruct->skyDropTargets[0] && gBattlerTarget != gBattleStruct->skyDropTargets[1] && gBattlerTarget != gBattleStruct->skyDropTargets[2] && gBattlerTarget != gBattleStruct->skyDropTargets[3])))
```

Comment:
again, this logic should be heavily simplified by my `skyDropTargets` suggestion

Reply:
If a Pokémon attempts to use Sky Drop on another Pokémon that is involved in a different instance of Sky Drop, the move will always fail, regardless of accuracy and abilities. That's what the added conditions to the if-statement handles.